Source/WebCore/accessibility/AXLogger.cpp

```diff
@@ -543,290 +543,12 @@ TextStream& operator<<(TextStream& stream, AXRelationType relationType)
-        stream << "AXRowCountChanged";
-        break;
-    case AXObjectCache::AXNotification::AXRowCollapsed:
-        stream << "AXRowCollapsed";
-        break;
-    case AXObjectCache::AXNotification::AXRowExpanded:
-        stream << "AXRowExpanded";
-        break;
-    case AXObjectCache::AXNotification::AXExpandedChanged:
-        stream << "AXExpandedChanged";
-        break;
-    case AXObjectCache::AXNotification::AXInvalidStatusChanged:
-        stream << "AXInvalidStatusChanged";
-        break;
-    case AXObjectCache::AXNotification::AXPressDidSucceed:
-        stream << "AXPressDidSucceed";
-        break;
-    case AXObjectCache::AXNotification::AXPressDidFail:
-        stream << "AXPressDidFail";
-        break;
-    case AXObjectCache::AXNotification::AXPressedStateChanged:
-        stream << "AXPressedStateChanged";
-        break;
-    case AXObjectCache::AXNotification::AXReadOnlyStatusChanged:
-        stream << "AXReadOnlyStatusChanged";
-        break;
-    case AXObjectCache::AXNotification::AXRequiredStatusChanged:
-        stream << "AXRequiredStatusChanged";
-        break;
-    case AXObjectCache::AXNotification::AXSortDirectionChanged:
-        stream << "AXSortDirectionChanged";
-        break;
-    case AXObjectCache::AXNotification::AXTextChanged:
-        stream << "AXTextChanged";
-        break;
-    case AXObjectCache::AXNotification::AXTextCompositionChanged:
-        stream << "AXTextCompositionChanged";
-        break;
-    case AXObjectCache::AXNotification::AXTextUnderElementChanged:
-        stream << "AXTextUnderElementChanged";
-        break;
-#if ENABLE(AX_THREAD_TEXT_APIS)
-    case AXObjectCache::AXNotification::AXTextRunsChanged:
-        stream << "AXTextRunsChanged";
-        break;
-#endif
-    case AXObjectCache::AXNotification::AXTextSecurityChanged:
-        stream << "AXTextSecurityChanged";
-        break;
-    case AXObjectCache::AXNotification::AXElementBusyChanged:
-        stream << "AXElementBusyChanged";
-        break;
-    case AXObjectCache::AXNotification::AXDraggingStarted:
-        stream << "AXDraggingStarted";
-        break;
-    case AXObjectCache::AXNotification::AXDraggingEnded:
-        stream << "AXDraggingEnded";
-        break;
-    case AXObjectCache::AXNotification::AXDraggingEnteredDropZone:
-        stream << "AXDraggingEnteredDropZone";
-        break;
-    case AXObjectCache::AXNotification::AXDraggingDropped:
-        stream << "AXDraggingDropped";
-        break;
-    case AXObjectCache::AXNotification::AXDraggingExitedDropZone:
-        stream << "AXDraggingExitedDropZone";
-        break;
-    case AXObjectCache::AXNotification::AXTextCompositionBegan:
-        stream << "AXTextCompositionBegan";
-        break;
-    case AXObjectCache::AXNotification::AXTextCompositionEnded:
-        stream << "AXTextCompositionEnded";
+#define WEBCORE_LOG_AXNOTIFICATION(name) \
+    case AXObjectCache::AXNotification::name: \
+        stream << #name; \
```

Comment:
If we put `“AX” #name` here then we could leave all the repeated AX out of the macro arguments.

Reply:
Great idea!